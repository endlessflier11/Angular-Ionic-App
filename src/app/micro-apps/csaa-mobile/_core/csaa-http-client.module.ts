import { NgModule } from '@angular/core';
import { AngularHttpAdapter } from './services/http/adapters/angular-http.adapter';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { IonicHttpAdapter } from './services/http/adapters/ionic-http.adapter';
import { HTTP } from '@ionic-native/http/ngx';
import { UNI_HTTP_ADAPTER } from './services/http/uni-http.model';

@NgModule({
  imports: [HttpClientModule],
  providers: [
    HTTP,
    { provide: UNI_HTTP_ADAPTER, multi: true, useClass: AngularHttpAdapter, deps: [HttpClient] },
    {
      provide: UNI_HTTP_ADAPTER,
      multi: true,
      useClass: IonicHttpAdapter,
      deps: [HTTP],
    },
  ],
  exports: [HttpClientModule],
})
export class CsaaHttpClientModule {}
