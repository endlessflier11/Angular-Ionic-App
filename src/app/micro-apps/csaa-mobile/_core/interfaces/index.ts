export * from './auth.interface';
export * from './router';
export * from './http.interface';
export * from './vehicle.interface';
export * from './ionic-lifecycle.interface';
export * from './analytics.interface';
export * from './policy.interface';
export * from './payment.interface';
export * from './config.interface';
export * from './ngxs-store.interface';
