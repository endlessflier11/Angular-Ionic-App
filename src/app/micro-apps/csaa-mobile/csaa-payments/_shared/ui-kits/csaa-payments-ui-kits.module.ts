import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaPaymentHistoryCardComponent } from './csaa-payment-history-card/csaa-payment-history-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentHistorySelectCardComponent } from './csaa-payment-history-select-card/csaa-payment-history-select-card.component';
import { CsaaPaymentSucceededCardComponent } from './csaa-payment-succeeded-card/csaa-payment-succeeded-card.component';
import { CsaaMakePaymentCardComponent } from './csaa-make-payment-card/csaa-make-payment-card.component';
import { CsaaPayAllPoliciesCardComponent } from './csaa-pay-all-policies-card/csaa-pay-all-policies-card.component';
import { PaymentTermsConditionsComponent } from './payment-terms-conditions/payment-terms-conditions.component';
import { AutopayTermsConditionsComponent } from './autopay-terms-conditions/autopay-terms-conditions.component';
import { CsaaBillsCardComponent } from './csaa-bills-card/csaa-bills-card.component';
import { AutopayDiscountCardComponent } from './autopay-discount-card/autopay-discount-card.component';
import { CsaaMakePaymentItemComponent } from './csaa-make-payment-item/csaa-make-payment-item.component';
import { IconPaymentMethodComponent } from './icon-payment-method/icon-payment-method.component';
import { CardSelectAmountComponent } from './card-select-amount/card-select-amount.component';
import { BrMaskerModule } from 'br-mask';
import { ReactiveFormsModule } from '@angular/forms';
import { CardGroupPaymentMethodComponent } from './card-group-payment-method/card-group-payment-method.component';

const SHARED_COMPONENTS = [
  CsaaPaymentHistoryCardComponent,
  CsaaPaymentHistorySelectCardComponent,
  CsaaMakePaymentCardComponent,
  CsaaPaymentSucceededCardComponent,
  CsaaPayAllPoliciesCardComponent,
  PaymentTermsConditionsComponent,
  AutopayTermsConditionsComponent,
  CsaaBillsCardComponent,
  AutopayDiscountCardComponent,
  CsaaMakePaymentItemComponent,
];

@NgModule({
  declarations: [
    ...SHARED_COMPONENTS,
    IconPaymentMethodComponent,
    CardSelectAmountComponent,
    CardGroupPaymentMethodComponent,
  ],
  imports: [
    CommonModule,
    IonicModule,
    UiKitsModule,
    IonicModule,
    BrMaskerModule,
    ReactiveFormsModule,
  ],
  exports: [
    ...SHARED_COMPONENTS,
    IconPaymentMethodComponent,
    CardSelectAmountComponent,
    CardGroupPaymentMethodComponent,
  ],
})
export class CsaaPaymentsUiKitsModule {}
