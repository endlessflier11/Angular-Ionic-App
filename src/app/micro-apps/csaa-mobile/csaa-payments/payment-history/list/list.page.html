<ion-header [ngClass]="['csaa', currentTheme]">
  <ion-toolbar>
    <ion-buttons slot="start">
      <csaa-back-button (back)="onClickBackBtn()"></csaa-back-button>
    </ion-buttons>
    <ion-buttons slot="end">
      <ion-button (click)="hoursPopup.show()">Call Service</ion-button>
    </ion-buttons>
  </ion-toolbar>
  <ion-toolbar>
    <ion-title>Transaction History</ion-title>
  </ion-toolbar>
</ion-header>
<ion-content [ngClass]="['csaa', currentTheme]">
  <ion-refresher slot="fixed" (ionRefresh)="doRefresh($event.target)" #refresher>
    <ion-refresher-content></ion-refresher-content>
  </ion-refresher>

  <!--  If user has multiple policy, select history to view -->
  <ng-container *ngIf="loading || (policies$ | async)?.length > 1">
    <csaa-payment-history-select-card
      [loading]="loading"
      [policies]="policies$ | async"
      (policyClick)="openPaymentHistory($event)"
    >
    </csaa-payment-history-select-card>
  </ng-container>

  <!-- If user has only one policy, display history for it -->
  <ng-container *ngIf="renderSinglePolicyHistoryCard( (history$|async), (policies$ | async) )">
    <csaa-payment-history-card
      [history]="(history$ | async)"
      [policy]="(policies$ | async)[0]"
      [loading]="loading"
    >
    </csaa-payment-history-card>
  </ng-container>

  <!-- Bills section -->
  <csaa-csaa-bills-card
    *ngIf="true"
    [loading]="loading || (isPaymentsLoading$ | async)"
    [bills]="latestBills$ | async"
    [autoPayEnrollmentStatusMap]="autoPayEnrollmentStatusMap"
    [isPaidInFullStatusMap]="isPaidInFullStatusMap"
    [availableDocumentsMap]="availableDocumentsMap"
    (billClick)="openBill($event)"
  >
  </csaa-csaa-bills-card>
</ion-content>
<csaa-working-hours #hoursPopup [ngClass]="['csaa', currentTheme]" serviceType="CUSTOMER">
</csaa-working-hours>
