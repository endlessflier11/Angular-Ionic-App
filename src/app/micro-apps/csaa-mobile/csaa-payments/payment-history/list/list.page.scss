csaa-payment-history-card {
  ::ng-deep {
    ion-card {
      ion-card-header {
        ion-item {
          margin-bottom: 16px;
        }
      }
    }
  }
}
