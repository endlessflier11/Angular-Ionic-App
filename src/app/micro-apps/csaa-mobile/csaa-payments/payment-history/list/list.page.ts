import { Component, OnDestroy, OnInit } from '@angular/core';
import { Select, Store } from '@ngxs/store';
import { ConfigState } from '../../../_core/store/states/config.state';
import { RouterService, PdfDisplayService, AnalyticsService } from '../../../_core/services';
import {
  PaymentHistory,
  Policy,
  AutoPayEnrollmentResponse,
  Bill,
  Category,
  EventName,
  EventType,
  UpcomingPayment,
} from '../../../_core/interfaces';
import { PaymentState } from '../../../_core/store/states/payment.state';
import { noop, Observable, concat } from 'rxjs';
import { PolicyState } from '../../../_core/store/states/policy.state';
import { SubSink } from '../../../_core/shared/subscription.helper';
import { CustomerAction, PaymentAction, PolicyAction } from '../../../_core/store/actions';
import { finalize, switchMap, tap } from 'rxjs/operators';
import { withErrorReporter, fetchHasFailedFor } from '../../../_core/helpers';
import { interactWithLoader } from '../../../_core/operators';
import { IonRefresher } from '@ionic/angular';
import { PolicyDocument } from '../../../_core/interfaces/document.interface';
import { FetchState } from '../../../_core/store/states/fetch.state';

@Component({
  selector: 'app-csaa-payment-history-list',
  templateUrl: './list.page.html',
  styleUrls: ['./list.page.scss'],
})
export class PaymentHistoryListPage implements OnInit, OnDestroy {
  @Select(PaymentState.paymentHistory) history$: Observable<PaymentHistory>;
  @Select(PolicyState.allPolicies) policies$: Observable<Policy[]>;
  @Select(PaymentState.upcomingPayments) payments$: Observable<UpcomingPayment[]>;
  @Select(PaymentState.latestBills) latestBills$: Observable<Bill[]>;
  @Select(FetchState.isLoading(PaymentAction.LoadPayments)) isPaymentsLoading$: Observable<boolean>;

  currentTheme: string;
  loading = false;

  // Special props for displaying bills
  autoPayEnrollmentStatusMap: { [policyNumber: string]: AutoPayEnrollmentResponse | null } = {};
  isPaidInFullStatusMap: { [policyNumber: string]: boolean } = {};
  availableDocumentsMap: { [policyNumber: string]: PolicyDocument[] } = {};

  private refresher: IonRefresher;
  private readonly subsink = new SubSink();

  constructor(
    private readonly store: Store,
    private readonly routerService: RouterService,
    private readonly analyticsService: AnalyticsService,
    private readonly pdfDisplayService: PdfDisplayService
  ) {
    this.currentTheme = this.store.selectSnapshot(ConfigState.theme);
  }

  ngOnInit(): void {
    this.loading = true;
    this.subsink.add(
      fetchHasFailedFor(
        this.store,
        CustomerAction.LoadCustomer,
        PolicyAction.LoadPolicies
      ).subscribe(() => this.routerService.navigateRoot()),
      this.payments$.subscribe((payments) => {
        // @ts-ignore
        this.autoPayEnrollmentStatusMap = Object.fromEntries(
          payments.map((p) => [p.policyNumber, p.autopayEnrollment || null])
        );
        // @ts-ignore
        this.isPaidInFullStatusMap = Object.fromEntries(
          payments.map((p) => [p.policyNumber, p.remainingPremium <= 0])
        );
      })
    );
    this.dispatchLoadActions();
  }

  ngOnDestroy(): void {
    this.subsink.unsubscribe();
  }

  invalidDateAndnoRemainder(payment: UpcomingPayment) {
    const today = new Date(Date.now());
    const dueDate = new Date(payment.dueDate);
    const diffDays = Math.floor((today.getTime() - dueDate.getTime()) / 1000 / 3600 / 24);

    const month = dueDate.getMonth() + 1;
    const year = dueDate.getFullYear();
    const daysInMonth = new Date(year, month, 0).getDate();
    return payment.remainingPremium <= 0 && diffDays >= daysInMonth;
  }

  onClickBackBtn() {
    this.routerService.back();
  }

  renderSinglePolicyHistoryCard(_history: PaymentHistory, policies: Policy[]) {
    return policies?.length === 1;
  }

  openPaymentHistory(policy) {
    this.routerService
      .navigateForward('csaa.payment.history.policy', { policyNumber: policy.number })
      .then(noop);
  }

  doRefresh(refresher) {
    this.loading = true;
    this.refresher = refresher;
    this.dispatchReloadActions();
  }

  openBill(bill: Bill) {
    this.store
      .dispatch(new PolicyAction.LoadPolicyDocuments(bill.policyNumber))
      .pipe(interactWithLoader())
      .subscribe(async () => {
        const documents = this.store.selectSnapshot(
          PolicyState.documentsForPolicy(bill.policyNumber)
        );
        // If Auto Pay is enrolled, use documentPolicyDocumentType.AutopaySchedule
        const documentFilterValue = this.autoPayEnrollmentStatusMap[bill.policyNumber]?.autoPay
          ? 'Auto Pay Schedule'
          : 'Billing';
        const documentFilterKey = this.autoPayEnrollmentStatusMap[bill.policyNumber]?.autoPay
          ? 'docName'
          : 'docType';
        // documents list already sorted at the api response level
        // Ref: src/app/micro-apps/csaa-mobile/_core/services/policy.service.ts:334
        // So we simply grab the first match
        const billDocument = documents.find((d) => d[documentFilterKey] === documentFilterValue);
        console.log({ billDocument, documents });
        if (billDocument) {
          this.pdfDisplayService.showDocumentOptions(billDocument);
        } else {
          const policy = this.store.selectSnapshot(PolicyState.policyData(bill.policyNumber));
          this.analyticsService.trackEvent(EventName.ERROR_NOTIFICATION, Category.documents, {
            event_type: EventType.MESSAGED,
            selection: 'Installment Bill',
            document_category: 'Billing & Payments',
            document_type: 'Billing Documents',
            document_name: 'Installment Bill',
            document_effective: 'current',
            process_date: bill.transactionDate,

            ...AnalyticsService.mapPolicy(policy),
          });
          this.pdfDisplayService.showDocumentErrorAlert();
        }
      });
  }

  private completeLoading() {
    this.loading = false;
    if (this.refresher) {
      this.refresher.complete().then(noop);
    }
  }

  private dispatchLoadActions() {
    this.store
      .dispatch(new CustomerAction.LoadCustomer())
      .pipe(
        switchMap(() => this.store.dispatch(new PolicyAction.LoadPolicies())),
        switchMap(() => this.store.dispatch(new PaymentAction.LoadPayments())),
        switchMap(() => this.store.dispatch(new PaymentAction.LoadHistory())),
        switchMap(() => {
          const sources: Observable<any>[] = [];
          // Fetch the available documents for all policies
          Object.entries(this.autoPayEnrollmentStatusMap).forEach(([policyNumber]) => {
            sources.push(
              this.store.dispatch(new PolicyAction.LoadPolicyDocuments(policyNumber)).pipe(
                tap((_) => {
                  this.availableDocumentsMap[policyNumber] = this.store.selectSnapshot(
                    PolicyState.documentsForPolicy(policyNumber)
                  );
                })
              )
            );
          });

          return concat(...sources);
        }),

        finalize(() => this.completeLoading())
      )
      .subscribe(withErrorReporter(noop));
  }

  private dispatchReloadActions() {
    this.store
      .dispatch(new CustomerAction.LoadCustomer())
      .pipe(
        switchMap(() => this.store.dispatch(new PolicyAction.ReloadPolicies())),
        switchMap(() => this.store.dispatch(new PaymentAction.ReloadPayments())),
        switchMap(() => this.store.dispatch(new PaymentAction.ReloadHistory())),
        finalize(() => this.completeLoading())
      )
      .subscribe(withErrorReporter(noop));
  }
}
