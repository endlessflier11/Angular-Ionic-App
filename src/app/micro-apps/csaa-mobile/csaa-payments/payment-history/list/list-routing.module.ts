import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { PaymentHistoryListPage } from './list.page';
import { AppRoutes } from '../../../_core/interfaces';

const routes: AppRoutes = [
  {
    isGroupIndex: true,
    path: '',
    component: PaymentHistoryListPage,
  },
];

@NgModule({
  declarations: [],
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
})
export class CsaaPaymentHistoryListRoutingModule {}
