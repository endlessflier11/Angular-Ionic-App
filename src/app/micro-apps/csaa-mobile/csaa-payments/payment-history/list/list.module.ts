import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IonicModule } from '@ionic/angular';
import { PaymentHistoryListPage } from './list.page';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentsUiKitsModule } from '../../_shared/ui-kits/csaa-payments-ui-kits.module';
import { CsaaPaymentHistoryListRoutingModule } from './list-routing.module';

@NgModule({
  declarations: [PaymentHistoryListPage],
  imports: [
    CommonModule,
    CsaaPaymentHistoryListRoutingModule,
    IonicModule,
    UiKitsModule,
    CsaaPaymentsUiKitsModule,
  ],
})
export class CsaaPaymentHistoryListModule {}
