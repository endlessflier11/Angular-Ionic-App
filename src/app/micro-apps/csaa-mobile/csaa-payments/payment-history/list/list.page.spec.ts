import { ComponentFixture, TestBed } from '@angular/core/testing';

import {
  By,
  CUSTOMER_STATE_FIXTURE_MOCK,
  DOCUMENTS_FIXTURE,
  PageTestingModule,
  setPolicyDocuments,
  StoreTestBuilder,
} from '@app/testing';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentsUiKitsModule } from '../../_shared/ui-kits/csaa-payments-ui-kits.module';
import { CallService } from '../../../_core/services/call.service';
import { CsaaAppInjector } from '../../../csaa-app.injector';
import { Injector } from '@angular/core';
import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
import { CsaaHttpClientModule } from '../../../_core/csaa-http-client.module';
import { CsaaCoreModule } from '../../../csaa-core/csaa-core.module';
import { Store } from '@ngxs/store';
import { Platform } from '@ionic/angular';
import { PaymentHistoryListPage } from './list.page';
import { USER1_STATE_FIXTURES_MOCK } from '../../../../../../testing/fixtures/state/by-user-state.fixture';
import { AppEndpointsEnum } from '../../../_core/interfaces';

const flushMockDocuments = (httpTestingController: HttpTestingController, policyDocuments) => {
  httpTestingController
    .match(AppEndpointsEnum[AppEndpointsEnum.policyDocuments])
    .forEach((req) => req.flush(policyDocuments));
};

describe('PaymentHistoryListPage', () => {
  let fixture: ComponentFixture<PaymentHistoryListPage>;
  let httpTestingController: HttpTestingController;
  let INITIAL_POLICES;
  let CUSTOMER;

  beforeEach(async () => {
    INITIAL_POLICES = CUSTOMER_STATE_FIXTURE_MOCK.csaa_policies.policies;
    CUSTOMER = INITIAL_POLICES.reduce(
      (customer, p) => setPolicyDocuments(customer, DOCUMENTS_FIXTURE, p.number),
      CUSTOMER_STATE_FIXTURE_MOCK
    );

    try {
      await TestBed.configureTestingModule({
        declarations: [PaymentHistoryListPage],
        imports: [
          PageTestingModule.withConfig({
            providesAnalytics: true,
            providesConfig: true,
            providesStorage: true,
          }),
          UiKitsModule,
          CsaaPaymentsUiKitsModule,
          CsaaCoreModule.forRoot(),
        ],
        providers: [
          HttpClientTestingModule,
          CsaaHttpClientModule,
          { provide: CallService, useValue: { call: jest.fn() } },
        ],
      }).compileComponents();

      CsaaAppInjector.injector = TestBed.inject(Injector);
      const store = TestBed.inject(Store);
      StoreTestBuilder.withDefaultMocks().withCustomerState(CUSTOMER).resetStateOf(store);

      await TestBed.inject(Platform).ready();

      httpTestingController = TestBed.inject(HttpTestingController);
      fixture = TestBed.createComponent(PaymentHistoryListPage);
    } catch (error) {
      console.error(error);
      fail(error.message);
    }
  });

  it('should create', () => {
    fixture.detectChanges();
    expect(fixture.componentInstance).toBeTruthy();
  });

  describe('user with multiple policies', () => {
    afterEach(() => {
      // Ensures no pending request
      httpTestingController.verify();
    });

    it('should match snapshot when content loaded', () => {
      fixture.detectChanges();
      flushMockDocuments(httpTestingController, DOCUMENTS_FIXTURE);

      fixture.detectChanges();
      const listComponentNode = fixture.debugElement.query(
        By.css('csaa-payment-history-select-card')
      );
      expect(listComponentNode).toBeTruthy();
      expect(listComponentNode.query(By.css('ion-list'))).toBeTruthy();
      expect(fixture).toMatchSnapshot();
    });
  });

  describe('user with one policy', () => {
    afterEach(() => {
      // Ensures no pending request
      httpTestingController.verify();
    });

    it('should match snapshot when content loaded', () => {
      const store = TestBed.inject(Store);
      StoreTestBuilder.withDefaultMocks()
        .withCustomerState(CUSTOMER)
        .withPolicyState({
          ...CUSTOMER.csaa_policies,
          policies: [USER1_STATE_FIXTURES_MOCK.POLICIES[0]],
        })
        .withPaymentState({
          ...CUSTOMER.csaa_payment,
          payments: [USER1_STATE_FIXTURES_MOCK.PAYMENTS[0]],
        })
        .resetStateOf(store);
      fixture = TestBed.createComponent(PaymentHistoryListPage);
      fixture.detectChanges();

      flushMockDocuments(httpTestingController, DOCUMENTS_FIXTURE);
      fixture.detectChanges();

      const listComponentNode = fixture.debugElement.query(By.css('csaa-payment-history-card'));
      expect(listComponentNode).toBeTruthy();
      expect(listComponentNode.query(By.css('ion-list'))).toBeTruthy();
      expect(fixture).toMatchSnapshot();
    });
  });
});
