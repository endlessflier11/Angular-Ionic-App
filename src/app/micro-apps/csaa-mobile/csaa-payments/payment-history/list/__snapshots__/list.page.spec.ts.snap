// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentHistoryListPage user with multiple policies should match snapshot when content loaded 1`] = `
<app-csaa-payment-history-list
  __history$__selector={[Function Observable]}
  __latestBills$__selector={[Function Observable]}
  __ngContext__={[Function LRootView]}
  __payments$__selector={[Function Observable]}
  __policies$__selector={[Function Observable]}
  analyticsService={[Function AnalyticsMockService]}
  autoPayEnrollmentStatusMap={[Function Object]}
  availableDocumentsMap={[Function Object]}
  currentTheme=""
  isPaidInFullStatusMap={[Function Object]}
  loading="false"
  pdfDisplayService={[Function PdfDisplayService]}
  routerService={[Function RouterService]}
  store={[Function Store]}
  subsink={[Function SubSink]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button>
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Back
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
      <ion-buttons
        slot="end"
      >
        <ion-button>
          Call Service
        </ion-button>
      </ion-buttons>
    </ion-toolbar>
    <ion-toolbar>
      <ion-title>
        Transaction History
      </ion-title>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <ion-refresher
      slot="fixed"
    >
      <ion-refresher-content />
    </ion-refresher>
    <csaa-payment-history-select-card>
      <ion-card
        class="csaa-card"
      >
        <ion-card-content
          class="csaa-item-list"
        >
          <ion-list>
            <ion-item
              class="ion-text-wrap"
              detail=""
            >
              <ion-label>
                <h3>
                  CAAC910026006
                </h3>
                <span
                  class="subtitle"
                >
                  <csaa-policy-subtitles>
                    <span
                      class="subtitle"
                    >
                      2009 Acura Tsx,
                    </span>
                    <span
                      class="subtitle"
                    >
                      2007 Jeep Grand Cherokee
                    </span>
                    <a
                      class="csaa-link"
                      href="#"
                    >
                      <small>
                        Show more
                      </small>
                    </a>
                  </csaa-policy-subtitles>
                </span>
              </ion-label>
              <csaa-theme-icon
                slot="end"
              >
                <img
                  src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                  style="width: 40px; height: 40px;"
                />
              </csaa-theme-icon>
            </ion-item>
            <ion-item
              class="ion-text-wrap"
              detail=""
            >
              <ion-label>
                <h3>
                  CAH3910026008
                </h3>
                <span
                  class="subtitle"
                >
                  <csaa-policy-subtitles>
                    <span
                      class="subtitle"
                    >
                      518 25th St
                    </span>
                  </csaa-policy-subtitles>
                </span>
              </ion-label>
              <csaa-theme-icon
                slot="end"
              >
                <img
                  src="/assets/csaa-mobile/vectors/icn-Home.svg"
                  style="width: 40px; height: 40px;"
                />
              </csaa-theme-icon>
            </ion-item>
            <ion-item
              class="ion-text-wrap"
              detail=""
            >
              <ion-label>
                <h3>
                  CAPU910026011
                </h3>
                <span
                  class="subtitle"
                >
                  <csaa-policy-subtitles>
                    <span
                      class="subtitle"
                    >
                      Personal Umbrella Policy
                    </span>
                  </csaa-policy-subtitles>
                </span>
              </ion-label>
              <csaa-theme-icon
                slot="end"
              >
                <img
                  src="/assets/csaa-mobile/vectors/icn-PUP.svg"
                  style="width: 40px; height: 40px;"
                />
              </csaa-theme-icon>
            </ion-item>
          </ion-list>
        </ion-card-content>
      </ion-card>
    </csaa-payment-history-select-card>
  </ion-content><csaa-working-hours
    class="csaa"
    servicetype="CUSTOMER"
  >
    <div
      class="csaa-working-hours hidden"
    >
      <ion-card
        class="card card-md"
      >
        <ion-card-header>
          Service Hours
        </ion-card-header>
        <ion-card-content
          class="card-content card-content-md"
        >
          <ion-list>
            <ion-item>
            </ion-item>
          </ion-list>
          <ion-button
            expand="full"
          >
             Call Service 
          </ion-button>
        </ion-card-content>
      </ion-card>
    </div>
  </csaa-working-hours>
</app-csaa-payment-history-list>
`;

exports[`PaymentHistoryListPage user with one policy should match snapshot when content loaded 1`] = `
<app-csaa-payment-history-list
  __history$__selector={[Function Observable]}
  __latestBills$__selector={[Function Observable]}
  __ngContext__={[Function LRootView]}
  __payments$__selector={[Function Observable]}
  __policies$__selector={[Function Observable]}
  analyticsService={[Function AnalyticsMockService]}
  autoPayEnrollmentStatusMap={[Function Object]}
  availableDocumentsMap={[Function Object]}
  currentTheme=""
  isPaidInFullStatusMap={[Function Object]}
  loading="false"
  pdfDisplayService={[Function PdfDisplayService]}
  routerService={[Function RouterService]}
  store={[Function Store]}
  subsink={[Function SubSink]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button>
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Back
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
      <ion-buttons
        slot="end"
      >
        <ion-button>
          Call Service
        </ion-button>
      </ion-buttons>
    </ion-toolbar>
    <ion-toolbar>
      <ion-title>
        Transaction History
      </ion-title>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <ion-refresher
      slot="fixed"
    >
      <ion-refresher-content />
    </ion-refresher>
    <csaa-payment-history-card>
      <ion-card
        class="csaa-card"
      >
        <ion-card-header
          class="policy-number-container"
        >
          <ion-item
            class="ion-text-wrap ion-no-padding csaa-policy-item"
          >
            <ion-label>
              <h3>
                CAAC910026006
              </h3>
              <p
                class="subtitle"
              >
                <csaa-policy-subtitles>
                  <span
                    class="subtitle"
                  >
                    2009 Acura Tsx,
                  </span>
                  <span
                    class="subtitle"
                  >
                    2007 Jeep Grand Cherokee
                  </span>
                  <a
                    class="csaa-link"
                    href="#"
                  >
                    <small>
                      Show more
                    </small>
                  </a>
                </csaa-policy-subtitles>
              </p>
            </ion-label>
            <csaa-theme-icon
              slot="end"
            >
              <img
                src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                style="width: 40px; height: 40px;"
              />
            </csaa-theme-icon>
          </ion-item>
        </ion-card-header>
        <ion-card-content
          class="csaa-item-list"
        >
          <ion-list>
          </ion-list>
        </ion-card-content>
      </ion-card>
    </csaa-payment-history-card>
  </ion-content><csaa-working-hours
    class="csaa"
    servicetype="CUSTOMER"
  >
    <div
      class="csaa-working-hours hidden"
    >
      <ion-card
        class="card card-md"
      >
        <ion-card-header>
          Service Hours
        </ion-card-header>
        <ion-card-content
          class="card-content card-content-md"
        >
          <ion-list>
            <ion-item>
            </ion-item>
          </ion-list>
          <ion-button
            expand="full"
          >
             Call Service 
          </ion-button>
        </ion-card-content>
      </ion-card>
    </div>
  </csaa-working-hours>
</app-csaa-payment-history-list>
`;
